### PR TITLE
Add kernel version, Docker version and awsvpc setting to checkpoint info

### LINKED ISSUE #2310: include docker and kernel version in checkpoint
This would help us make decision about version compatibility.

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #2310 

I couldn't find any library anywhere that did the necessary kernel wrangling, until after I wrote my own, when I found https://github.com/docker/docker/blob/master/pkg/platform/.  I don't like the way they report errors unconditionally via `logrus` so didn't use it.

(No errors are documented for `uname(2)` except "bad pointer" which can't happen here)

### CODE REVIEW

**Comment by awh** [comment]
I propose we add a new entry at the very top of `site/operational-guide/tasks.md` named 'Disable checkpoint' which describes the data we collect, and how to switch it off.

**Comment by rade** [comment]
Ops guide seems a strange place. I would have though this info belongs in the "Checkpoint" section of https://www.weave.works/docs/net/latest/installing-weave/

**Comment by awh** [comment]
It can go there too

**Comment by rade** [comment]
It shouldn't go in two places. And "Installing Weave" is the better of the two, imo.

**Comment by awh** [comment]
LGTM; just needs docs

**Comment by bboreham** [comment]
Added a section to the docs re checkpoint info

File: site/installing-weave.md [hunk 1]
```diff
@@ -36,6 +36,12 @@ check, run the following before launching Weave Net:
 
     export CHECKPOINT_DISABLE=1
 
+The information sent in this check is:
```
Comment (brb) [inline site/installing-weave.md]: @abuehrle This requires your attention.

File: site/installing-weave.md [hunk 2]
```diff
@@ -36,6 +36,12 @@ check, run the following before launching Weave Net:
...
+
+ * Weave current version
```
Comment (rade) [inline site/installing-weave.md]: "Weave Net" (and I'd drop the 'current')

File: site/installing-weave.md [hunk 3]
```diff
@@ -36,6 +36,12 @@ check, run the following before launching Weave Net:
...
+ * Kernel version
+ * Docker version
+ * AWSVPC mode
```
Comment (rade) [inline site/installing-weave.md]: "Network mode, e.g. 'awsvpc'"

Comment (bboreham) [inline site/installing-weave.md]: Note we don't send any other modes at present.

Comment (rade) [inline site/installing-weave.md]: I know.

File: site/installing-weave.md (hunk 1 above)
Comment (rade) [inline site/installing-weave.md]: Surely we should also mention the host UUID.

**Comment by abuehrle** [comment]
LGTM

File: site/installing-weave.md (hunk 1 above)
Comment (bboreham) [inline site/installing-weave.md]: We don't send that. We send a hash, which I didn't unpick.

Comment (rade) [inline site/installing-weave.md]: "host UUID hash"? "anonymized host UUID"

Basically I think it's important to communicate that we can tie the information to a single, albeit anonymised, host.

**Comment by rade** [comment]
I have addressed the remaining points.

**Pull request merged**